### PR TITLE
nfs_version: allow to manage multiple versions of NFS.

### PR DESCRIPTION
**Pull request by lion24**
It could be rather usefull for this role to also allow managing NFS3
share.

Our use case require some options that are not supported in NFSv4, so
allow to specify nfsvers options could be rather handy :-)

Signed-off-by: Lionel H <lionel.hubaut@tessares.net>

### CODE REVIEW

**Comment by joshmoore** [comment]
Thanks @lion24 . Would you mind updating the variable to match the `nfs_` prefix? I'd probably suggest the somewhat long `nfs_version` as well. If that works for you, we'll likely add also a disclaimer that we don't regularly test this version, unless you can see a way to add molecule tests for nfsv3.

**Comment by lion24** [comment]
Sure @joshmoore will update the variable name to match `nfs_` prefix ;)
Regarding molecule test, that's something I didn't dig into it. So you can, for now, add a disclaimer regarding the NFSv3 version.

File: tasks/main.yml [hunk 1]
```diff
@@ -22,9 +22,9 @@
 - name: Mount NFS share
   become: yes
   mount:
-    fstype: nfs4
+    fstype: nfs
     name: "{{ item.path }}"
-    opts: "{{ item.opts | default(nfs_mount_opts) }}"
+    opts: "vers={{ nfs_version }},{{ item.opts | default(nfs_mount_opts) }}"
```
Comment (joshmoore) [inline tasks/main.yml:27]: Having seen `nfsvers` elsewhere, adding a comment here for my peace of mind:

> **`nfsvers=n`**
> The NFS protocol version number used to contact the server's NFS service. If the server does not support the requested version, the mount request fails. If this option is not specified, the client negotiates a suitable version with the server, trying version 4 first, version 3 second, and version 2 last.
>
> **`vers=n`**
> This option is an alternative to the nfsvers option. It is included for compatibility with other operating systems
```

**Comment by joshmoore** [comment]
Proposing for release as 1.1.3 by EOB.

**Review by sbesson — APPROVED** [review]
Generally looks good. As a side comment, this also reflects the fact `nfs4` is actually deprecated in `/etc/fstab` - http://manpages.ubuntu.com/manpages/xenial/man5/nfs.5.html, https://linux.die.net/man/5/nfs.

The main impact for a consumer upgrading the role is that `/etc/fstab` will be flagged as changed in the playbook execution. However since the expectation is that functionally the configuration line should be identical, no objection to having this as a patch release.

**Comment by joshmoore** [comment]
> nfs4 is actually deprecated in /etc/fstab

:+1:

> /etc/fstab will be flagged as changed

Shall we go for 1.2.0 then?

**Comment by sbesson** [comment]
Given that this PR adds a backwards-compatible API to set the `nfs_version`, `1.2.0` might be the most unambiguous version bump.

**Comment by joshmoore** [comment]
Sounds like a deal. Releasing. Thanks again @lion24 !

**Comment by joshmoore** [comment]
1.2.0 is now available on https://galaxy.ansible.com/openmicroscopy/nfs-mount

**Comment by lion24** [comment]
@joshmoore Great ! It was a pleasure to make this PR :smiley: 

Cheers.

**Comment by joshmoore** [comment]
Glad to hear it!

**Pull request merged**